### ISSUE TITLE
Failed rollover proposal due to missing announcement does not update taker UI state

### ISSUE THREAD

**Issue by thomaseizinger** [itchysats/itchysats#360]
1. Have an open CFD
2. Propose rollover
3. Rollover accepted
4. Announcement not ready yet, rollover fails
5. UI does not update

![image](https://user-images.githubusercontent.com/5486389/137674731-22f3fe20-055c-4e8b-86cd-90e798aed418.png)

**Comment by da-kami** [comment]
This was on the taker side (I was the maker), but was the error of `Announcement not ready yet` observed on the taker side as well? 
Isn't this already solved with https://github.com/comit-network/hermes/pull/362 ?

**Comment by bonomat** [comment]
It looks like there is something wrong with the order of things on the maker side: 

I added some more log statements: 
```
2021-10-22 14:51:39 DEBUG Maker accepts a roll_over proposal order_id=4d5faed0-4ca2-4ce0-8681-1db7e42f99db
2021-10-22 14:51:39 ERROR Message handler failed: Announcement /x/BitMEX/BXBT/2021-10-23T04:00:00.price?n=20 not found
2021-10-22 14:51:44 DEBUG Fetching announcement for /x/BitMEX/BXBT/2021-10-23T04:00:00.price?n=20 from https://h00.ooo/x/BitMEX/BXBT/2021-10-23T04:00:00.price?n=20
2021-10-22 14:51:44 DEBUG Received new announcement: NewAnnouncementFetched
```

The error is thrown here: https://github.com/comit-network/hermes/blob/f5db38a7242f5189bcf96e1898eeb69e7e9a6064/daemon/src/maker_cfd.rs#L752

**Comment by bonomat** [comment]
Ah, understood. This is by design. If `self.announcements` does not contain that event, it will trigger fetching it but in the meantime returns `None`.

**Comment by bonomat** [comment]
PR #419 solves the first issue of not having the needed announcements for being able to roll over.

**Comment by bonomat** [comment]
Imho this should have been solved with #419. Let's close and see if it happens again.
